pkg/validate/validate_component_test.go

```diff
@@ -173,21 +92,33 @@ components:
     componentName: foo-comp
     version: 2.10.1
     appVersion: 3.10`,
-		},
-		{
-			desc: "success: X.Y.Z-blah app version",
-			set:  defaultComponentSetNoRefs,
-			components: `
-components:
-- apiVersion: 'bundle.gke.io/v1alpha1'
-  kind: Component
-  metadata:
-    name: foo-comp-1.0.2
-  spec:
-    componentName: foo-comp
-    version: 2.10.1
-    appVersion: 3.10.32-blah.0`,
-		},
+		},*/
+
+func assertValidateComponent(componentString string, numErrors int, t *testing.T){
+	component, _ := converter.FromYAMLString(componentString).ToComponent()
+	errs := Component(component)
+	numErrorsActual := len(errs)
+	if numErrorsActual != numErrors {
+		t.Fatalf("expected %v errors got %v", numErrors, numErrorsActual)
+	}
+}	
+
+func TestComponentVersionDotNotationAppVersion(t *testing.T){
```

Comment:
why split up the table test?

Reply:
Personally I would prefer to have the refactoring the tests in a separate PR regardless. Table tests are pretty standard go-style, so I'm not sure what we gain from this  (TestComponentVersionDotNotationAppVersion feels very Java-y).

Reply:
https://github.com/golang/go/wiki/TableDrivenTests

Reply:
So again, didn't mean to push this yet, but the old method All function was convoluting ideas as far as I could tell.  I was separating them out and pulling them out into individual tests was just easier.